pkg/cmd/gitprovider/add.go

```diff
@@ -44,9 +49,84 @@ var GitProviderAddCmd = &cobra.Command{
 		setGitProviderConfig.Username = new(string)
 		setGitProviderConfig.Alias = new(string)
 
-		err = gitprovider_view.GitProviderCreationView(ctx, apiClient, &setGitProviderConfig, existingAliases)
-		if err != nil {
-			return err
+		if len(args) == 0 {
+			err = gitprovider_view.GitProviderCreationView(ctx, apiClient, &setGitProviderConfig, existingAliases)
+			if err != nil {
+				return err
+			}
+		} else {
+			supportedProviders := config.GetSupportedGitProviders()
+			supportedProviderIds := util.ArrayMap(supportedProviders, func(gp config.GitProvider) string {
+				return gp.Id
+			})
+
+			if slices.Contains(supportedProviderIds, args[0]) {
+				setGitProviderConfig.ProviderId = args[0]
+			}
+			providerId := setGitProviderConfig.ProviderId
+
+			if providerId == "" {
+				supportedProvidersStr := strings.Join(supportedProviderIds, ", ")
+				return fmt.Errorf("'%s' is invalid or not a supported git provider.\nSupported providers are: %s", args[0], strings.TrimSuffix(supportedProvidersStr, ", "))
```

Comment:
No need for the `TrimSuffix` here. 
https://go.dev/play/p/tUFuH4aVLzE

Reply:
fixed